lib/parachain/types.go

```diff
@@ -0,0 +1,388 @@
+	PersistedValidationDataHash common.Hash `scale:"4"`
+
+	// PovHash is the hash of the `pov-block`.
+	PovHash common.Hash `scale:"5"`
+	// ErasureRoot is the root of a block's erasure encoding Merkle tree.
+	ErasureRoot common.Hash `scale:"6"`
+
+	// Signature on blake2-256 of components of this receipt:
+	// The parachain index, the relay parent, the validation data hash, and the `pov_hash`.
+	// this is basically sr25519::Signature
+	Signature CollatorSignature `scale:"7"`
+
+	// ParaHead is the hash of the para header that is being generated by this candidate.
+	ParaHead common.Hash `scale:"8"`
+	// ValidationCodeHash is the blake2-256 hash of the validation code bytes.
+	ValidationCodeHash ValidationCodeHash `scale:"9"`
+}
+
+// OccupiedCore Information about a core which is currently occupied.
+type OccupiedCore struct {
+	// NOTE: this has no ParaId as it can be deduced from the candidate descriptor.
+	// If this core is freed by availability, this is the assignment that is next up on this
+	// core, if any. None if there is nothing queued for this core.
+	NextUpOnAvailable *ScheduledCore `scale:"1"`
+	// The relay-chain block number this began occupying the core at.
+	OccupiedSince types.BlockNumber `scale:"2"`
+	// The relay-chain block this will time-out at, if any.
+	TimeoutAt types.BlockNumber `scale:"3"`
+	// If this core is freed by being timed-out, this is the assignment that is next up on this
+	// core. None if there is nothing queued for this core or there is no possibility of timing
+	// out.
+	NextUpOnTimeOut *ScheduledCore `scale:"4"`
+	// A bitfield with 1 bit for each validator in the set. `1` bits mean that the corresponding
+	// validators has attested to availability on-chain. A 2/3+ majority of `1` bits means that
+	// this will be available.
+	Availability []byte `scale:"5"`
+	// The group assigned to distribute availability pieces of this candidate.
+	GroupResponsible GroupIndex `scale:"6"`
+	// The hash of the candidate occupying the core.
+	CandidateHash common.Hash `scale:"7"`
+	// The descriptor of the candidate occupying the core.
+	CandidateDescriptor CandidateDescriptor `scale:"8"`
+}
+
+// Index returns the index
+func (OccupiedCore) Index() uint {
+	return 0
+}
+
+// ScheduledCore Information about a core which is currently occupied.
+type ScheduledCore struct {
+	// The ID of a para scheduled.
+	ParaID ParaID
+	// The collator required to author the block, if any.
+	Collator *Collator
+}
+
+// Index returns the index
+func (ScheduledCore) Index() uint {
+	return 1
+}
+
+// Free Core information about a core which is currently free.
+type Free scale.VaryingDataType
+
+// Index returns the index
+func (Free) Index() uint {
+	return 2
+}
+
+// CoreState represents the state of a particular availability core.
+type CoreState scale.VaryingDataType
+
+// AvailabilityCores represents a slice of CoreState
+type AvailabilityCores scale.VaryingDataTypeSlice
```

Comment:
Im wondering if for here and for CandidateEvents (and others if I missed them) if we should have a function to set the value when using it in exports.go.

Kinda like how we do [here](https://github.com/ChainSafe/gossamer/blob/85ebb2c9a4b60ab23006e80e31bd10ca05bb19c1/dot/types/digest.go#L24) but where we take a param.

Not sure if this is needed/preferred but putting it here as a thought